### PR TITLE
BUG: correct mb03rd for default X=None argument

### PR DESCRIPTION
**Pull request by roryyorke**
Added a test for this case.

@repagh I think this is a reasonably simple fix -- do you have time to review it?

### CODE REVIEW

**Comment by roryyorke** [comment]
Failing in the Python 2.7 builds because I used @ in the test; also failing in all conda recipe builds, no sure why. 

Here's an import failure in from Ubuntu Bionic / Python 3.8 / AMD64 / conda / OpenBLAS (https://travis-ci.org/github/python-control/Slycot/jobs/752107811):

```
ImportError: /home/travis/miniconda/conda-bld/slycot_1609325503642/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/python3.8/site-packages/slycot/_wrapper.cpython-38-x86_64-linux-gnu.so: undefined symbol: npy_PyErr_ChainExceptionsCause
```

I don't think this is caused by this PR, which doesn't change build settings.

**Comment by roryyorke** [comment]
2.7 now passing.

**Review by bnavigator — APPROVED** [review]

File: slycot/math.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def mb03rd(n, A, X=None, jobx='U', sort='N', pmax=1.0, tol=0.0):
             The matrix `A` to be block-diagonalized, in real Schur form.
         X : (n, n) array_like, optional
             A given matrix `X`, for accumulation of transformations (only if
-            `jobx`='U')
+            `jobx`='U').   Default value is identity matrix of order `n`.
```
Comment (bnavigator) [inline slycot/math.py:45]:
```suggestion
            `jobx`='U'). Default value is identity matrix of order `n`.
```

File: slycot/tests/test_mb.py [hunk 2]
```diff
@@ -86,6 +86,27 @@ def test_mb03rd(self):
                     test1_n, A, X, 'N', sort, test1_pmax, test1_tol)
             assert Xr is None
 
+
```
Comment (bnavigator) [inline slycot/tests/test_mb.py:89]:
```suggestion
```

File: slycot/tests/test_mb.py [hunk 3]
```diff
@@ -86,6 +86,27 @@ def test_mb03rd(self):
...
+        X = Tschur.copy()
+
+        Ar, Xr, blsize, W = mb03rd(Aschur.shape[0], Aschur, X, 'U', 'N', pmax=1.0, tol=0.0)
+
+        Ar2, Xr2, blsize2, W2 = mb03rd(Aschur.shape[0], Aschur)
+
+        assert_allclose(Ar, Ar2)
+        assert_allclose(Xr, Tschur.dot(Xr2))
+
```
Comment (bnavigator) [inline slycot/tests/test_mb.py:108]:
```suggestion
```

**Review by bnavigator — COMMENTED** [review]
Some whitespace style review comments (mainly to retrigger Travis)

**Pull request merged**